packages/backend-deployer/src/cdk_deployer.ts

```diff
@@ -199,14 +199,23 @@ export class CDKDeployer implements BackendDeployer {
       // However if the cdk process didn't run or produced no output, then we have nothing to go on with. So we throw
       // this error to aid in some debugging.
       if (aggregatedStderr.trim()) {
+        // If the string is more than 65KB, truncate and keep the last portion.
         // eslint-disable-next-line amplify-backend-rules/prefer-amplify-errors
-        throw new Error(aggregatedStderr);
+        throw new Error(this.truncateString(aggregatedStderr, 65000));
       } else {
         throw error;
       }
     }
   };
 
+  private truncateString = (str: string, size: number) => {
+    const encoder = new TextEncoder();
+    const decoder = new TextDecoder();
+    const truncated = encoder.encode(str).slice(-size); // truncate the beginning
+
+    return decoder.decode(truncated);
```

Comment:
I'd suggest to simplify this and just take 65k characters instead of bytes.

The reason is that:
1. it seems node is using UTF-8 encoding for strings by default.
2. this transformation is transforming it back and forth to `Uint8Array`.

That seems that effectively "number of bytes" is "number of characters". (or perhaps relationship is at least constant in case default encoding is different than what I mentioned above).

Reply:
That's what I had it in the beginning but learned that UTF-8 is [actually 1-4 bytes](https://www.ibm.com/docs/en/db2-for-zos/12?topic=unicode-utfs), and the VARCHAR used in DB counts bytes and not characters.

Reply:
I see.
Could you check if `Buffer.byteLength(string, encoding)` could be used here?
It may be a bit faster and avoid some data copies.